pulp_2_tests/tests/rpm/api_v2/test_modularity.py

```diff
@@ -430,3 +431,71 @@ def test_all(self):
             RPM_UNSIGNED_FEED_COUNT - sum(MODULE_FIXTURES_PACKAGES.values()),
             non_modular_units
         )
+
+
+class CheckModulesYamlTestCase(unittest.TestCase):
+    """Check whether modules.yaml is available if appropriate."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class wide variables."""
+        cls.cfg = config.get_config()
+        if cls.cfg.pulp_version < Version('2.18'):
+            raise unittest.SkipTest('This test requires Pulp 2.18 or newer.')
+        cls.client = api.Client(cls.cfg, api.json_handler)
+
+    def test_no_modules_yaml_generated_non_modular(self):
+        """Verify no modules yaml is generated for non modular content.
+
+        This test does the following
+
+        1. Create and sync a non-modular content.
+        2. Publish the synced content.
+        3. Check whether no modules.yaml is generated for published content.
+
+        This test case targets:
+
+        * `Pulp #4350 <https://pulp.plan.io/issues/4350>`_.
+        """
+        body = gen_repo(
+            importer_config={'feed': RPM_UNSIGNED_FEED_URL},
+            distributors=[gen_distributor(auto_publish=True)]
+        )
+        # Step 1 and 2
+        repo = self.client.post(REPOSITORY_PATH, body)
+        self.addCleanup(self.client.delete, repo['_href'])
+        sync_repo(self.cfg, repo)
+        repo = self.client.get(repo['_href'], params={'details': True})
+        # Step 3
+        files = self.list_repo_data_files(self.cfg, repo)
+        # check no modules.yaml.gz is found
+        self.assertFalse(bool(files))
+        modules_elements = self.get_modules_elements_repomd(
+            self.cfg,
+            repo['distributors'][0]
+        )
+        self.assertFalse(bool(modules_elements))
+
+    @staticmethod
+    def list_repo_data_files(cfg, repo):
+        """Return a list of all the files present inside repodata dir."""
+        return cli.Client(cfg).run((
+            'find',
```

Comment:
What about add `-type f ` to assure that just files are returned?

Reply:
Have Made the changes.